src/fsharp/service/service.fs

```diff
@@ -279,9 +279,12 @@ type BackgroundCompiler(
                                 match ilReaderOpt with
                                 | Some ilReader ->
                                     let ilModuleDef, ilAsmRefs = ilReader.ILModuleDef, ilReader.ILAssemblyRefs
-                                    return RawFSharpAssemblyData(ilModuleDef, ilAsmRefs) :> IRawFSharpAssemblyData |> Some
+                                    let data = RawFSharpAssemblyData(ilModuleDef, ilAsmRefs) :> IRawFSharpAssemblyData
+                                    return ProjectAssemblyDataResult.Available data
                                 | _ ->
-                                    return None
+                                    // Note 'false' - if a PEReference doesn't find an ILModuleReader then we don't
+                                    // continue to try to use an on-disk DLL
+                                    return ProjectAssemblyDataResult.Unavailable false
```

Comment:
Ok, so this should behave the same as before. - We basically only do `useOnDisk` for type providers.

Reply:
Yes, though I suspect it's right for this path too - it's hard to imagine a situation where dropping the reference on the floor is better than reusing an on-disk DLL